### ISSUE TITLE
Strange behaviour of an empty item (but with whitespace between start/end tags) in List

### ISSUE THREAD

**Issue by Hronom** [FasterXML/jackson-dataformat-xml#191]
Strange behaviour of empty item in list (v2.7.3)

So I have next xml file `TestList.xml`:
```xml
<TestList>
    <items>
        <item name="Item1"/>
        <item name="Item2">
        </item>
        <item name="Item3"/>
        <item name="Item4"/>
        <item name="Item5"/>
    </items>
</TestList>
```

But when I deserializing this file, I lost all items after item with name `Item2`, so items: `Item3`, `Item4`, `Item5` is lost. If I change this xml file to next format:
```xml
<TestList>
    <items>
        <item name="Item1"/>
        <item name="Item2"/>
        <item name="Item3"/>
        <item name="Item4"/>
        <item name="Item5"/>
    </items>
</TestList>
```
All start working as expected.

In version `2.6.3` all work as expected...

# Code listing
## `pom.xml`
```xml
        <!-- Jackson -->
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-annotations</artifactId>
            <version>2.7.3</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.7.3</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.7.3</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.dataformat</groupId>
            <artifactId>jackson-dataformat-xml</artifactId>
            <version>2.7.3</version>
        </dependency>
```

## `App.java`
```java
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.xml.XmlMapper;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class App {
    public static void main(String[] args) throws IOException {
        Path testListPath = Paths.get("TestList.xml");
        ObjectMapper mapper = new XmlMapper();
        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
        TestList testList = mapper.readValue(Files.newInputStream(testListPath), TestList.class);
        System.out.println();
    }
}
```

## `TestList.java`
```java
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;

import java.util.ArrayList;

public class TestList {
    @JacksonXmlElementWrapper(useWrapping = true, localName = "items")
    @JacksonXmlProperty(localName = "item")
    public ArrayList<ListItem> items = new ArrayList<>();
}
```

## `ListItem.java`
```java
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;

public class ListItem {
    @JacksonXmlProperty(isAttribute = true)
    public String name;
}
```

**Comment by cowtowncoder** [comment]
Looks like a bug.

I suspect this is because entry with `Item2` has both attribute value and textual content. Although textual content is "empty" from human perspective, from XML perspective it has distinctive non-empty value (that is, String length above 0).

**Comment by Hronom** [comment]
@cowtowncoder Maybe some kind of flag that can give different behaviours, Is there any flag to achieve behaviour like in `2.6.3` ?

**Comment by cowtowncoder** [comment]
@Hronom this is not intentional, so I don't think there is any need to support existing behavior. Problem is more with figuring out to fix the problem. Did this behave differently with 2.6.3?

**Comment by Hronom** [comment]
@cowtowncoder Yes when I change dependencies to `2.6.3`:
```xml
        <!-- Jackson -->
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-annotations</artifactId>
            <version>2.6.3</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.6.3</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.6.3</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.dataformat</groupId>
            <artifactId>jackson-dataformat-xml</artifactId>
            <version>2.6.3</version>
        </dependency>
```
All works as expected.

**Comment by cowtowncoder** [comment]
@Hronom thank you for additional information. This must then be a regression due to fixes included in 2.7; handling of empty Strings is a tricky problem.

**Comment by cowtowncoder** [comment]
One thing on test: it's usually a good idea NOT to disable "fail on unknown properties". Enabling it often masks legitimate failures. Not a big deal here, just thought I'll mention it.

**Comment by Hronom** [comment]
@cowtowncoder thanks for your advice, I think I enable it on real project.

**Comment by cowtowncoder** [comment]
@Hronom right, there are cases where it's useful to disable it, but you can also just disable it for specific types with `@JsonIgnoreProperties(ignoreUnknown=true)`.

I can reproduce this issue now with 2.7.3. Thanks!

**Comment by cowtowncoder** [comment]
Guessing this might be due fix for #177, which would also affect 2.6.5.

**Comment by cowtowncoder** [comment]
Was able to fix this with minor modification to #177 fix; and looks like this also took care of #178.
